ecl/hthor/hthor.cpp

```diff
@@ -8226,58 +8227,22 @@ bool CHThorDiskReadBaseActivity::openNext()
-                        if (actualDiskMeta)
-                        {
-                            translator.setown(createRecordTranslator(projectedDiskMeta->queryRecordAccessor(true), actualDiskMeta->queryRecordAccessor(true)));
-                            if (translator->needsTranslate())
-                            {
-                                keyedTranslator.setown(createKeyTranslator(actualDiskMeta->queryRecordAccessor(true), expectedDiskMeta->queryRecordAccessor(true)));
-                                if (translator->canTranslate())
-                                {
-                                    if (agent.rltEnabled()==RecordTranslationMode::None)
-                                    {
-    #ifdef _DEBUG
-                                        translator->describe();
-    #endif
-                                        throw MakeStringException(0, "Translatable key layout mismatch reading file %s but translation disabled", logicalFileName.str());
-                                    }
-                                }
-                                else
-                                    throw MakeStringException(0, "Untranslatable key layout mismatch reading file %s", logicalFileName.str());
-                            }
-                            else
-                                translator.clear();  // MORE - could question why the format appeared to mismatch
-                        }
-                        else
-                            throw MakeStringException(0, "Untranslatable key layout mismatch reading file %s - key layout information not found", logicalFileName.str());
-                    }
-                    else
-                    {
-                        actualDiskMeta.set(helper.queryDiskRecordSize()->querySerializedDiskMeta());
-                        if (grouped)
-                            actualDiskMeta.setown(new CSuffixedOutputMeta(+1, actualDiskMeta));
-                    }
-                }
-            }
+                actualDiskMeta.setown(getDaliLayoutInfo(props));
+                actualCrc = props.getPropInt("@formatCrc");
+            }
+            if (!actualDiskMeta)
+                actualDiskMeta.set(expectedDiskMeta->querySerializedDiskMeta());
+            keyedTranslator.setown(createKeyTranslator(actualDiskMeta->queryRecordAccessor(true), expectedDiskMeta->queryRecordAccessor(true)));
+            if (keyedTranslator && keyedTranslator->needsTranslate())
+                keyedTranslator->translate(actualFilter, fieldFilters);
             else
-            {
-                translator.clear();
-                keyedTranslator.clear();
-            }
-            calcFixedDiskRecordSize();
+                actualFilter.appendFilters(fieldFilters);
+
+            bool canSerialize = actualDiskMeta->queryTypeInfo()->canSerialize() && projectedDiskMeta->queryTypeInfo()->canSerialize();
```

Comment:
minor: canSerializeTypeInfo would be a better indication of what couldn't be serialized.

Reply:
will change.